printThis.js

```diff
@@ -92,7 +92,7 @@
                 $body = $doc.find("body");
 
             // add base tag to ensure elements use the parent domain
-            $head.append('<base href="' + document.location.protocol + '//' + document.location.host + '">');
+            $head.append('<base href="' + document.location.protocol + '//' + document.location.host + document.location.pathname+'">');
```

Comment:
I think that if your path would be /blog/entry/if-i-was-a-rich-man your css-path would be relative to that path. ie:
../../mycss.css

But you are correct it will not always work. But could a option-parameter be an option ?
So I could provide my own base-string ? That would be sweet :)